src/App.js

```diff
@@ -30,21 +32,97 @@ class App extends React.Component {
   };
 
   // Insert form callback functions handleChange and handleSubmit here
+  handleSubmit = (event) => {
+    event.preventDefault();
+    if (this.state.guessValue.length !== 1) {
+      this.setState({ inputWarning: <p>Only one letter is allowed</p> });
+    } else if (this.state.guessedLetters.includes(this.state.guessValue)) {
+      this.setState({
+        inputWarning: <p>You have already guessed this letter</p>,
+      });
+    } else {
+      this.setState({
+        guessedLetters: [...this.state.guessedLetters, this.state.guessValue],
+        guessValue: "",
+        inputWarning: null,
+      });
+    }
```

Comment:
Without too many if/else spaghetti we can make this a bit more readable by returning early. Might not be a big improvement here, but rather a general guideline you could follow.

```suggestion
    if (this.state.guessValue.length !== 1) return this.setState({ inputWarning: <p>Only one letter is allowed</p> });
    
    if (this.state.guessedLetters.includes(this.state.guessValue)) {
      return this.setState({
        inputWarning: <p>You have already guessed this letter</p>,
      });
    }
    
    this.setState({
        guessedLetters: [...this.state.guessedLetters, this.state.guessValue],
        guessValue: "",
        inputWarning: null,
      });
    }
```